rct229/rulesets/ashrae9012019/section11/section11rule14.py

```diff
@@ -0,0 +1,137 @@
+                        "$.service_water_heating_distribution_systems[*].id", rmd_b
+                    )
+                    for swh_dist_sys_id_b in swh_comps_dict_b[swh_dist_id_b].uses
+                    for swh_use_p in find_all(
+                        f'$.buildings[*].building_segments[*].zones[*].spaces[*].service_water_heating_uses[*][?(@.id="{swh_dist_sys_id_b}")].use',
+                        rmd_p,
+                    )
+                ]
+            )
+
+            return swh_use_loads_p
+
+        def create_data(self, context, data):
+            rmd_b = context.BASELINE_0
+
+            swh_comps_dict_b = (
+                get_swh_equipment_associated_with_each_swh_distribution_system(rmd_b)
+            )
+
+            piping_info_b = {}
+            for swh_dist_sys_b in find_all(
+                "$.service_water_heating_distribution_systems[*]", rmd_b
+            ):
+                for piping_id_b in swh_comps_dict_b[swh_dist_sys_b["id"]].piping:
+                    piping = find_one_with_field_value(
+                        "$.service_water_heating_distribution_systems[*].service_water_piping[*]",
+                        "id",
+                        piping_id_b,
+                        rmd_b,
+                    )
+                    piping_id = piping["id"]
+                    piping_info_b[piping_id] = {}
+                    piping_info_b[piping_id]["IS_RECIRC"] = piping.get(
+                        "is_recirculation_loop"
+                    )
+
+                    if piping_info_b[piping_id]["IS_RECIRC"]:
+                        for pump_id in swh_comps_dict_b[swh_dist_sys_b["id"]].pumps:
+                            pump = find_one_with_field_value(
+                                "$.pumps[*]",
+                                "id",
+                                pump_id,
+                                rmd_b,
+                            )
+                            piping_info_b[piping_id]["PUMP_POWER"] = pump.get(
+                                "design_electric_power"
+                            )
+
+            return {
+                "swh_comps_dict_b": swh_comps_dict_b,
+                "piping_info_b": piping_info_b,
+            }
+
+        class SWHDistributionRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section11Rule14.RMDRule.SWHDistributionRule, self).__init__(
+                    rmds_used=produce_ruleset_model_description(
+                        USER=False, BASELINE_0=True, PROPOSED=True
+                    ),
+                )
+
+            def manual_check_required(self, context, calc_vals=None, data=None):
+                piping_info_b = data["piping_info_b"]
+
+                return any(not piping["IS_RECIRC"] for piping in piping_info_b.values())
+
+            def get_calc_vals(self, context, data=None):
+                piping_info_b = data["piping_info_b"]
+
+                return {"piping_info_id_b": piping_info_b}
```

Comment:
The type of ```piping_info_id_b``` is dict[str, dict]. What would be the best way to pass this dict over to the ```rule_check``` considering the reporting?

Reply:
Addressed.